### PR TITLE
Make package name link to package documentation page

### PR DESCRIPTION
**Pull request by Themanwhosmellslikesugar**

### CODE REVIEW

File: src/modules/Builder/components/PackageList/PackageList.jsx [hunk 1]
```diff
@@ -18,12 +70,17 @@ export default class PackageList extends Component {
     }
 
     render() {
+        const { basePackagesUrl } = this.state;
+        const { os, device } = this.props;
+
         return (
-            <PackagesTable key="packages-list-table" />
+            <PackagesTable key="packages-list-table" device={device} os={os} basePackagesUrl={basePackagesUrl} />
```
Comment (karpa4o4) [inline src/modules/Builder/components/PackageList/PackageList.jsx:77]: No need to pass `device`;

File: src/modules/Builder/components/PackageList/components/PackagesTable/PackagesTable.jsx [hunk 2]
```diff
@@ -13,6 +15,7 @@ import Input from "common/components/Input";
 import Select from "common/components/Select";
 
 import Blackmagic from "api/rpc/blackmagic";
+
```
Comment (karpa4o4) [inline src/modules/Builder/components/PackageList/components/PackagesTable/PackagesTable.jsx:18]: This is an unnecessary change.

File: src/modules/Builder/components/PackageList/components/PackagesTable/PackagesTable.jsx [hunk 3]
```diff
@@ -66,6 +69,7 @@ export default class PackagesTable extends Component {
         this.onPageChange = this.onPageChange.bind(this);
         this.onSearchFieldChange = this.onSearchFieldChange.bind(this);
         this.getCardFooter = this.getCardFooter.bind(this);
+        this.packageLink = this.packageLink.bind(this);
```
Comment (karpa4o4) [inline src/modules/Builder/components/PackageList/components/PackagesTable/PackagesTable.jsx:72]: This `bind` is unnecessary.

File: src/modules/Builder/components/PackageList/PackageList.jsx [hunk 4]
```diff
@@ -2,14 +2,66 @@ import React, { Component } from "react";
...
+        }
+
+        return title.trim();
+    }
+
     constructor(props) {
         super(props);
 
+        this.state = {
+            basePackagesUrl: "",
```
Comment (karpa4o4) [inline src/modules/Builder/components/PackageList/PackageList.jsx:29]: Let's set `null` by default.

File: src/modules/Builder/components/PackageList/PackageList.jsx [hunk 5]
```diff
@@ -2,14 +2,66 @@ import React, { Component } from "react";
...
         this.blackmagic = new Blackmagic();
+
+        this.getBasePackagesUrl = this.getBasePackagesUrl.bind(this);
+    }
+
+    componentDidMount() {
+        API.listDevice()
+            .then((response) => {
+                this.setState(() => ({
+                    basePackagesUrl: this.getBasePackagesUrl(response.data),
```
Comment (karpa4o4) [inline src/modules/Builder/components/PackageList/PackageList.jsx:41]: This binding is unnecessary.

File: src/modules/Builder/components/PackageList/PackageList.jsx [hunk 6]
```diff
@@ -2,14 +2,66 @@ import React, { Component } from "react";
...
     constructor(props) {
         super(props);
 
+        this.state = {
+            basePackagesUrl: "",
+        };
+
         this.blackmagic = new Blackmagic();
+
+        this.getBasePackagesUrl = this.getBasePackagesUrl.bind(this);
```
Comment (karpa4o4) [inline src/modules/Builder/components/PackageList/PackageList.jsx:34]: This binding is unnecessary.

File: src/modules/Builder/components/PackageList/components/PackagesTable/PackagesTable.jsx [hunk 7]
```diff
@@ -167,6 +171,7 @@ export default class PackagesTable extends Component {
         return "unset";
     }
 
+
```
Comment (karpa4o4) [inline src/modules/Builder/components/PackageList/components/PackagesTable/PackagesTable.jsx:174]: This is an unnecessary change.

File: src/modules/Builder/components/PackageList/components/PackagesTable/PackagesTable.jsx [hunk 8]
```diff
@@ -248,6 +253,22 @@ export default class PackagesTable extends Component {
         return packageObj;
     }
 
+    packageLink(currentPackage) {
+        const { basePackagesUrl, os } = this.props;
+        let link = basePackagesUrl;
```
Comment (karpa4o4) [inline src/modules/Builder/components/PackageList/components/PackagesTable/PackagesTable.jsx:258]: Define this variable on 264 line.

File: src/modules/Builder/components/PackageList/components/PackagesTable/PackagesTable.jsx [hunk 9]
```diff
@@ -248,6 +253,22 @@ export default class PackagesTable extends Component {
...
+
+        if (!link) {
+            return false;
+        }
+
+        link += currentPackage.package;
```
Comment (karpa4o4) [inline src/modules/Builder/components/PackageList/components/PackagesTable/PackagesTable.jsx:264]: You can use the template string here and on 2656line.

File: src/modules/Builder/components/PackageList/components/PackagesTable/PackagesTable.jsx [hunk 10]
```diff
@@ -248,6 +253,22 @@ export default class PackagesTable extends Component {
         return packageObj;
     }
 
+    packageLink(currentPackage) {
+        const { basePackagesUrl, os } = this.props;
+        let link = basePackagesUrl;
+
+        if (!link) {
+            return false;
```
Comment (karpa4o4) [inline src/modules/Builder/components/PackageList/components/PackagesTable/PackagesTable.jsx:261]: Return null or undefined, but not boolean value.

File: src/modules/Builder/components/PackageList/PackageList.jsx [hunk 11]
```diff
@@ -2,14 +2,66 @@ import React, { Component } from "react";
...
+            currentOs = currentDevice.os.find((element) => (
+                os === element.short_name
+            ));
+        }
+
+        if (currentOs) {
+            return currentOs.base_packages_url;
+        }
+
+        return "";
```
Comment (karpa4o4) [inline src/modules/Builder/components/PackageList/PackageList.jsx:64]: Return `null` or nothing, then the value will be `undefined`.

File: src/modules/Builder/components/PackageList/PackageList.jsx (hunk 4 above)
Comment (Themanwhosmellslikesugar) [reply src/modules/Builder/components/PackageList/PackageList.jsx:29]: if set `null` in console we will see `Failed prop type: The prop `basePackagesUrl` is marked as required in `PackagesTable`, but its value is null.`

Comment (karpa4o4) [reply src/modules/Builder/components/PackageList/PackageList.jsx:29]: Let's make basePackagesUrl not requeired and `null` by default.
